packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -47,33 +48,18 @@ export async function getFileLink(
         return maybeFileLinkCacheEntry;
     }
 
-    const valueGenerator = async function() {
-        let result: string | undefined;
-        let success = false;
-        let retryAfterMs = 1000;
-        do {
-            try {
-                result = await getFileLinkCore(getToken, odspUrlParts, identityType, logger);
-                success = true;
-            } catch (err) {
-                // If it is not retriable, then just throw
-                if (!canRetryOnError(err)) {
-                    // Delete from the cache to permit retrying later.
-                    fileLinkCache.delete(cacheKey);
```

Comment:
Do we still want this `fileLinkCache.delete`? Seems like the behavior may still be desirable.

Reply:
Good catch. I added this behavior back with another commit.